tez-mapreduce/src/main/java/org/apache/tez/mapreduce/hadoop/MRJobConfig.java

```diff
@@ -131,6 +131,11 @@ public interface MRJobConfig {
 
   public static final String CACHE_ARCHIVES_VISIBILITIES = "mapreduce.job.cache.archives.visibilities";
 
+  /**
+   * Used by committers to set a job-wide UUID.
+   */
+  public static final String JOB_COMMITTER_UUID = "job.committer.uuid";
```

Comment:
This is not the setting used by s3 committer right? How will it work ?

Reply:
There is a corresponding change I have in my Hadoop code where it will consult this property similar to how it consults the property Spark sets for this purpose.

Reply:
so you can confirm this will work with job.committer.uuid, right?
can you link that point in hadoop code for later reference?

Reply:
@VenkatSNarayanan May i ask that if the hadoop s3 committer can work with Hive+Tez after this change?
IMO, s3/magic committer can avoid some operation like rename on s3, which can speed up/improve hive job.

Reply:
@abstractdog I haven't publicly posted the Hadoop PR yet, but the change I have is to check for this property around here: https://github.com/apache/hadoop/blob/51cb858cc8c23d873d4adfc21de5f2c1c22d346f/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/AbstractS3ACommitter.java#L1372 similar to how the Spark property is checked. I have tested these changes together already alongside my Hive implementation.

@zhangbutao There are some corresponding changes to Hadoop and Hive that also need to be merged which I have. Once all 3 PRs(Tez, Hadoop and Hive have been merged), then the magic committer will be usable with Hive.

Reply:
Let's hold of committing this in Tez till the Hadoop patch gets reviewed and committed. If there is a setting name change requested there, this will have to change again.

Reply:
Good point. It doesn't seem like there are any other major concerns about the Tez design so I will put up the Hadoop PR next.

Reply:
should this go into Tez 0.10.4? if so, it would be good to have it in 1-2 weeks, just FYI, about planning the hadoop change

Reply:
0.10.4 would be ideal. In that case, let me loop in the Hadoop folks to see if they have any strong opinions about this.

Reply:
There haven't been any objections from the Hadoop folks, I think it should be safe to go ahead with the patch as it is @abstractdog .